gobblin-runtime/src/main/java/org/apache/gobblin/runtime/template/InheritingJobTemplate.java

```diff
@@ -103,13 +132,14 @@ public Config getRawTemplateConfig() throws SpecNotFoundException, TemplateExcep
   private Config getRawTemplateConfigHelper(Set<JobTemplate> alreadyInheritedTemplates)
       throws SpecNotFoundException, TemplateException {
     Config rawTemplate = getLocalRawTemplate();
-    for (JobTemplate template : Lists.reverse(this.superTemplates)) {
-      if (!alreadyInheritedTemplates.contains(template)) {
-        alreadyInheritedTemplates.add(template);
-        Config thisFallback = template instanceof InheritingJobTemplate
-            ? ((InheritingJobTemplate) template).getRawTemplateConfigHelper(alreadyInheritedTemplates)
-            : template.getRawTemplateConfig();
-        rawTemplate = rawTemplate.withFallback(thisFallback);
+    if (this.superTemplates != null) {
```

Comment:
Why did all these `superTemplates` null checks have to be added? What is the case where it will be null?

Reply:
It needs to be added because this change essentially gives resolution process an interface to provide loaded templates and ask for resolution, instead of providing template URI and ask for resolution as it was. This is used for determining if the physical templates are provided.